core/src/main/java/org/elasticsearch/search/highlight/HighlightBuilder.java

```diff
@@ -289,7 +346,95 @@ public static HighlightBuilder fromXContent(QueryParseContext parseContext) thro
         return highlightBuilder;
     }
 
+    public void parse(XContentParser parser, SearchContext context) throws Exception {
```

Comment:
We shouldn't have this in the final version since the XContent will always be parsed to a HighlighterBuilder on the coordinating node and never parsed and passed straight to the SearchContext. If this is needed right now before that work has been done could you add a NORELEASE comment to mark it for removal?